benchmarks/src/main/java/org/elasticsearch/benchmark/index/mapper/BeatsMapperBenchmark.java

```diff
@@ -0,0 +1,205 @@
+
+    private static SourceToParse[] readSampleDocuments() throws IOException, URISyntaxException {
+        return Files.readAllLines(pathToResource("sample_documents.json"))
+            .stream()
+            .map(source -> new SourceToParse(UUIDs.randomBase64UUID(), new BytesArray(source), XContentType.JSON))
+            .toArray(SourceToParse[]::new);
+    }
+
+    private SourceToParse[] generateRandomDocuments(int count) {
+        var docs = new SourceToParse[count];
+        for (int i = 0; i < count; i++) {
+            docs[i] = generateRandomDocument();
+        }
+        return docs;
+    }
+
+    private SourceToParse generateRandomDocument() {
+        return new SourceToParse(
+            UUIDs.randomBase64UUID(),
+            new BytesArray(
+                "{    \"@timestamp\": "
+                    + System.currentTimeMillis()
+                    + ",    \"log.file.path\": \""
+                    + randomFrom("logs-1.log", "logs-2.log", "logs-3.log")
+                    + "\",    \"log.level\": \""
+                    + "INFO"
+                    + "\",    \"log.logger\": \""
+                    + "some.package.for.logging.requests"
+                    + "\",    \"client.ip\": \""
+                    + randomIp()
+                    + "\",    \"http.request.method\": \""
+                    + randomFrom("GET", "POST")
+                    + "\",    \"http.request.id\": \""
+                    + random.nextInt()
+                    + "\",    \"http.request.bytes\": "
+                    + random.nextInt(1024)
+                    + ",    \"url.path\": \""
+                    + randomString(1024)
+                    + "\",    \"http.response.status_code\": "
+                    + randomFrom(200, 204, 300, 404, 500)
+                    + ",    \"http.response.bytes\": "
+                    + random.nextInt(1024)
+                    + ",    \"http.response.mime_type\": \""
+                    + randomFrom("application/json", "application/xml")
+                    + "\"}"
+            ),
+            XContentType.JSON
+        );
+    }
+
+    private String randomIp() {
+        return "" + random.nextInt(255) + '.' + random.nextInt(255) + '.' + random.nextInt(255) + '.' + random.nextInt(255);
+    }
+
+    private String randomString(int maxLength) {
+        var length = random.nextInt(maxLength);
+        var builder = new StringBuilder(length);
+        for (int i = 0; i < length; i++) {
+            builder.append((byte) (32 + random.nextInt(94)));
+        }
+        return builder.toString();
+    }
+
+    @SafeVarargs
+    @SuppressWarnings("varargs")
+    private <T> T randomFrom(T... items) {
+        return items[random.nextInt(items.length)];
+    }
+
+    private static Path pathToResource(String path) throws URISyntaxException {
+        return Paths.get(BeatsMapperBenchmark.class.getResource(path).toURI());
+    }
+
+    private static MapperService createMapperService(String mappings) {
```

Comment:
I think we can add a TODO here at least, asking for this to be dried up with similar logic in the benchmark we copied this from?